x/trigger/keeper/gas_limit_test.go

```diff
@@ -0,0 +1,179 @@
+package keeper_test
+
+import (
+	"github.com/provenance-io/provenance/x/trigger/types"
+)
+
+func (s *KeeperTestSuite) TestGetAllGasLimits() {
+	tests := []struct {
+		name      string
+		gasLimits []types.GasLimit
+	}{
+		{
+			name:      "valid - no gas limits",
+			gasLimits: []types.GasLimit(nil),
+		},
+		{
+			name: "valid - one gas limit",
+			gasLimits: []types.GasLimit{
+				{
+					TriggerId: 1,
+					Amount:    5,
+				},
+			},
+		},
+		{
+			name: "valid - multiple gas limits",
+			gasLimits: []types.GasLimit{
+				{
+					TriggerId: 1,
+					Amount:    5,
+				},
+				{
+					TriggerId: 2,
+					Amount:    7,
+				},
+			},
+		},
+	}
+
+	for _, tc := range tests {
+		s.Run(tc.name, func() {
+			for _, gasLimit := range tc.gasLimits {
+				s.app.TriggerKeeper.SetGasLimit(s.ctx, gasLimit.TriggerId, gasLimit.Amount)
+			}
+			gasLimits, err := s.app.TriggerKeeper.GetAllGasLimits(s.ctx)
+			s.NoError(err)
+			s.Equal(tc.gasLimits, gasLimits)
+		})
+	}
+}
+
+func (s *KeeperTestSuite) TestGetAndSetGasLimits() {
+	s.app.TriggerKeeper.SetGasLimit(s.ctx, 1, 5)
+
+	tests := []struct {
+		name     string
+		id       types.TriggerID
+		expected uint64
+		panic    bool
+	}{
+		{
+			name:     "valid - gas limit",
+			id:       1,
+			expected: 5,
+			panic:    false,
+		},
+		{
+			name:     "invalid - gas limit doesn't exist",
+			id:       999,
+			expected: 0,
+			panic:    true,
+		},
+	}
+
+	for _, tc := range tests {
+		s.Run(tc.name, func() {
+			if tc.panic {
+				s.Panics(func() {
```

Comment:
Similar to the `.Error(...)` comment above, you should almost always be using `.PanicsWithError` or `.PanicsWithValue` so you're making sure that the panic is coming from what you expect it to come from. It's really easy for a test to panic due to NPE or index out of range and have it go missed because the reason for the panic is being ignored.